test/NuGet.Core.Tests/NuGet.Commands.Test/WarningPropertiesCollectionTests.cs

```diff
@@ -720,5 +739,44 @@ public void WarningPropertiesCollection_MessageWithTargetGraphAndDependencyWithN
             Assert.True(warningPropertiesCollection.ApplyWarningProperties(suppressedMessage));
             Assert.Equal(0, suppressedMessage.TargetGraphs.Count);
         }
+
+        [Theory]
+        [MemberData(nameof(WarningsNotAsErrorsTestCases))]
+        public void ApplyWarningProperties_WithProjectWarningsNotAsErrors(bool allWarningsAsErrors, HashSet<NuGetLogCode> noWarnSet, HashSet<NuGetLogCode> warnAsErrorSet, HashSet<NuGetLogCode> warningsNotAsErrors, LogLevel expectedLogLevel, bool suppressed)
+        {
+            // Arrange
+            var warningPropertiesCollection = new WarningPropertiesCollection(new WarningProperties(warnAsErrorSet, noWarnSet, allWarningsAsErrors, warningsNotAsErrors), null, null);
+            var logMessage = new RestoreLogMessage(LogLevel.Warning, NuGetLogCode.NU1503, "Warning");
+
+            // Act && Assert
+            bool result = warningPropertiesCollection.ApplyWarningProperties(logMessage);
+
+            // Assert
+            result.Should().Be(suppressed);
+            logMessage.Level.Should().Be(expectedLogLevel);
+        }
+
+        // WarningsAsErrors, NoWarn, WarnAsError, WarnNotAsError, ExpectedLogCode, Supressed
+        public static IEnumerable<object[]> WarningsNotAsErrorsTestCases()
```

Comment:
This contains 9 test cases, and as far as I can tell, the first 7 are about testing precedence of NoWarn, WarnAsError and WarnNotAsError. I think the last 2 are testing that codes specified in the properties, but are not actually raised during restore, so not change the result.  I think from a "single responsibility principals" point of view, the tests should be separated and have more meaningful test names.